internal/models/task.go

```diff
@@ -0,0 +1,67 @@
+package models
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os/exec"
+	"strings"
+
+	"github.com/thatnerdjosh/example-webservices/internal/config"
+)
+
+type TaskRequest struct {
+	Name string `json:"name"`
+}
+
+var (
+	ErrTaskNotFound = errors.New("task was not found")
+	ErrBadData      = errors.New("malformed data provided")
+)
+
+func (t *TaskRequest) Process(cfg *config.TaskConfig, r *http.Request) error {
+	b, _ := io.ReadAll(r.Body)
+	defer r.Body.Close()
+
+	err := t.load(b)
+	if err != nil {
+		return err
+	}
+
+	return t.run(cfg)
+}
+
+func (t *TaskRequest) load(input []byte) error {
+	err := json.Unmarshal(input, &t)
+	if t.Name == "" {
+		err = ErrBadData
```

Comment:
This should wrap the unmarshaling error - the error handling is a serious area of improvement for this code.